### DISCUSSION TITLE
Customization of the Authorization Header in authentication.py

### DISCUSSION THREAD

**Discussion by HtmlMak** [encode/django-rest-framework#9463] [Potential Issue]
In the file authentication.py of the Django REST Framework, located at [this line](https://github.com/encode/django-rest-framework/blob/master/rest_framework/authentication.py#L20), it would be beneficial to introduce the ability to customize the authorization header.

Currently, the authorization header is hardcoded, which can be limiting in scenarios where the application is secured with additional basic authentication, such as in test environments. By allowing the authorization header to be configurable, it would provide greater flexibility for various use cases and simplify the process of adjusting the header as needed.

**Comment by browniebroke** [comment]
I may have misunderstood what you're asking for, but there is another class for basic auth: 

https://github.com/encode/django-rest-framework/blob/ccfe0a963799f52c137a7ed555648e7bbffa0d93/rest_framework/authentication.py#L53-L59

If you need something more advanced/custom, it should be pretty simple to write your own auth class, you should make sure it extends the `BaseAuthentication` class which define the interface you need to follow:

https://github.com/encode/django-rest-framework/blob/ccfe0a963799f52c137a7ed555648e7bbffa0d93/rest_framework/authentication.py#L33-L50

**Reply by HtmlMak** [reply]
Yes, I did just that. The issue is that I had to completely copy the code of the authenticate method only because get_authorization_header is outside the class and cannot be overridden.

To provide some context: my application uses `TokenAuthentication`, which by default relies on the `Authorization` header. This works well in production. However, our test environments are secured with additional basic authentication by nginx to prevent unauthorized access.


My solution:
![image](https://github.com/encode/django-rest-framework/assets/9065989/3223f6d7-f55b-4940-b0e9-f7a26d0a788a)

My idea:
Change `request.META.get("HTTP_AUTHORIZATION", b"")` to `request.META.get(settings.AUTH_HEADER, b"")`
